infrastructure/router/router.go

```diff
@@ -0,0 +1,17 @@
+package router
+
+import (
+	"github.com/jvillarreal-w/academy-go-q42021/interface/controller"
+	"github.com/labstack/echo"
+	"github.com/labstack/echo/middleware"
+)
+
+func NewRouter(e *echo.Echo, c controller.AppController) *echo.Echo {
+	e.Use(middleware.Logger())
+	e.Use(middleware.Recover())
+
+	e.GET("/pokemon", func(context echo.Context) error { return c.Pokemon.GetPokemon(context) })
```

Comment:
I'm not familiar with echo, but would it be possible to just have, for example: 
`e.GET("/pokemon", c.Pokemon.GetPokemon(context))` ?

Just to avoid redundancy

Reply:
Same for the route below

Reply:
Seems like the intended way of declaring routes with Echo, though to be fair I cannot say I am familiar with this either.
https://echo.labstack.com/guide/routing/